jax/_src/numpy/lax_numpy.py

```diff
@@ -2910,6 +2920,18 @@ def delete(arr, obj, axis=None):
   # Case 3: obj is an array
   # NB: pass both arrays to check for appropriate error message.
   util.check_arraylike("delete", arr, obj)
+
+  # Case 3a: unique integer indices; delete in a JIT-compatible way
+  if issubdtype(_dtype(obj), integer) and assume_unique_indices:
```

Comment:
We could change this to
```
if issubdtype(_dtype(obj), integer) and (assume_unique_indices or np.size(obj) == 1):
```
to catch the traced-integer-index automatically, what do you think? The only change in behavior would be that out-of-bound static indices would be clipped instead of erroring as they do currently.

Reply:
(Instead of `np.size(obj) == 1`, maybe `obj.shape == ()`? The latter is a bit more dynamic-shape friendly, and also accepting `obj = jnp.array([[[[5]]]])` down this path seems weird.)

Hrm, iiuc we'd need this to get to the #12592 "desired state", right?

Changing the error behavior seems bad/surprising if the user isn't expecting it, but maybe `assume_unique_indices` can be (in principle) `assume_unique_inbounds_indices`? Or we could have a separate `assume_inbounds_indices`?

Reply:
Well, `obj` here might be a scalar or another object without a `shape` attribute, because at this point we don't know whether to cast it to a jax array or a numpy array. What about `np.shape(obj) == ()`?